SU2_CFD/include/solvers/CEulerSolver.hpp

```diff
@@ -1084,6 +1086,35 @@ class CEulerSolver : public CFVMFlowSolverBase<CEulerVariable, ENUM_REGIME::COMP
    */
   void InitTurboContainers(CGeometry *geometry, CConfig *config) final;
 
+  /*!
+   * \brief Initilize turbomachinery performance shared_ptr.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] config - Definition of the particular problem.
+   */
+  void InitTurboPerformance(CGeometry *geometry, CConfig *config) final;
+
+    /*!
+   * \brief Get Primal variables for turbo performance computation
+   *        iteration can be executed by multiple threads.
+   * \return returns Density, pressure and TurboVelocity (IN/OUTLET)
+   */
+  inline vector<su2double> GetTurboPrimitive(unsigned short iBlade, unsigned short iSpan, bool INLET) {
+    TurboPrimitive.clear();
```

Comment:
Where does this TurboPrimitive live? The solver? We have been cutting down on that type of aux member var.
In this particular case it may even be more efficient to create it locally (because of "return value optimization")